README.md

```diff
@@ -1,73 +1,68 @@
 # Breaking your Rust code for fun & profit
 
-[![Build Status](https://travis-ci.org/llogiq/mutagen.svg?branch=master)](https://travis-ci.org/llogiq/mutagen)
-[![Downloads](https://img.shields.io/crates/d/mutagen.svg?style=flat-square)](https://crates.io/crates/mutagen/)
-[![Version](https://img.shields.io/crates/v/mutagen.svg?style=flat-square)](https://crates.io/crates/mutagen/)
-[![License](https://img.shields.io/crates/l/mutagen.svg?style=flat-square)](https://crates.io/crates/mutagen/)
+*this is a architecture-preview, not all components are there*
 
-This is a work in progress mutation testing framework. Not all components are there, those that are there aren't finished, but it's already somewhat usable as of now.
+This is a mutation testing framework for Rust code.
 
-### Mutation Testing
+## Mutation Testing
 
-The idea behind mutation testing is to insert changes into your code to see if they make your tests fail. If not, your tests obviously fail to test the changed code.
-The difference to line or branch coverage is that those measure if the code under test was *executed*, but that says nothing about whether the tests would have caught any error.
-
-This repo has three components at the moment: The mutagen test runner, a helper library and a procedural macro that mutates your code.
-
-### How mutagen Works
-
-Mutagen works as a procedural macro. This means two things:
-
-1. You'll need a nightly rust toolchain to compile the plugin.
-2. it only gets to see the code you mark up with the `#[mutate]` annotation, nothing more.
+A change (mutation) in the program source code is most likely a bug of some kind. A good test suite can detect changes in the source code by failing ("killing" the mutant). If the test suite is green even if the program is mutated (the mutant survives), the tests fail to detect this bug.
 
-It also will only see the bare AST, no inferred types, no control flow or data flow, unless we analyse them ourselves. But not only that, we want to be *fast*.  This means we want to avoid doing one compile run per mutation, so we try to bake in all mutations into the code once and select them at runtime via a mutation count. This means we must avoid mutations that break the code so it no longer compiles.
```

Comment:
Why are we losing this information? I feel it is a crucial piece to understand the design decisions for mutagen, which haven't really changed with your redesign.

Reply:
This information is indeed important. I thought I kept this information somewhere. I will add this paragraph back into the documentation and the readme.

Reply:
In the newest commits this section is back in the readme. The documents about the architecture and design decisions also contain paragraphs on this topic.